rubix-tests/src/test/java/com/qubole/rubix/tests/TestThriftServerJVM.java

```diff
@@ -179,4 +184,49 @@ public void testCreateBookKeeperClient_OnNotRunningPort() throws IOException, In
     String host = "localhost";
     bookKeeperFactory.createBookKeeperClient(host, conf);
   }
+
+  public void testBookKeeperHealth() throws IOException, InterruptedException, TTransportException, TException
+  {
+    String host = "localhost";
+
+    RetryingBookkeeperClient client;
+    client = bookKeeperFactory.createBookKeeperClient(host, conf);
+
+    boolean isBookKeeperAlive = client.isBookKeeperAlive();
+    assertTrue(isBookKeeperAlive == true, "BookKeeper is not Alive");
+  }
+
+  @Test(enabled = true, expectedExceptions = org.apache.thrift.shaded.transport.TTransportException.class)
+  public void testBookKeeperHealthAtWrongPort() throws IOException, InterruptedException, TTransportException, TException
+  {
+    String host = "localhost";
+
+    //1234 is a random port to test if BookKeeperHealth is detecting that bookkeeper is down or not
+    CacheConfig.setServerPort(conf, 1234);
+    RetryingBookkeeperClient client;
+    client = bookKeeperFactory.createBookKeeperClient(host, conf);
+
+    boolean isBookKeeperAlive = client.isBookKeeperAlive();
+  }
+
+  @Test(enabled = true)
+  public void testBookKeeperHealthMain() throws IOException, InterruptedException, TTransportException, TException
+  {
```

Comment:
lets have either of this or the other test method you have written above. Both are doing same thing except this one is launing bookkeeper daemon in a separate process.